### PR TITLE
Switch from github.com/golang/mock to go.uber.org/mock

### PR DESCRIPTION
**Pull request by aruiz14**
## Issue: rancher/rancher#46972

- Adds `go:generate` commands for `pkg/generic/fake`, so there is no need to generate it manually anymore.
- Refresh generated files
- Update imports to the new dependency.
- `go mod tidy`

### CODE REVIEW

File: pkg/generic/fake/README.md [hunk 1]
```diff
@@ -65,16 +65,9 @@ mock.EXPECT().List(gomock.Any()).Return(nil, nil)
 ```
 
 ## Fake Generation
-This package was generated with `mockgen -package fake -destination ./controller.go -source ../controller.go` and `mockgen -package fake -destination ./cache.go -source ../cache.go`
+This package was generated with `mockgen`, see [`generate.go`](./generate.go).
```
Comment (ericpromislow) [inline pkg/generic/fake/README.md:68]: Is it possible to include the command that was run to generate this file?

I believe this option is in the uber fork of this library

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Just a quick comment. The rest is fine.

File: pkg/generic/fake/README.md (hunk 1 above)
Comment (aruiz14) [reply pkg/generic/fake/README.md]: The fork will automatically add a hint of the `mockgen` command used to generate it, but it's not really customizable.
Nonetheless, the only command needed must be `go generate` (just added it to the README). There is no need for users/developers to run `mockgen` command directly.

**Review by ericpromislow — APPROVED** [review]

**Review by moio — APPROVED** [review]
Switching to a maintained version is a good thing (tm)

approved

**Pull request merged**